src/ray/pubsub/publisher.cc

```diff
@@ -359,13 +372,15 @@ bool Publisher::RegisterSubscription(const rpc::ChannelType channel_type,
   return subscription_index_it->second.AddEntry(key_id.value_or(""), subscriber);
 }
 
-void Publisher::Publish(const rpc::PubMessage &pub_message) {
+void Publisher::Publish(rpc::PubMessage pub_message) {
+  RAY_CHECK(pub_message.sequence_id() == 0) << "sequence_id should not be set;";
```

Comment:
use RAY_CHECK_EQ so that it prints out `pub_message.sequence_id()` when the check fails